### PR TITLE
lock user out after 5 minutes of inactivity

### PR DESCRIPTION
**Pull request by wadeking98**
Signed-off-by: wadeking98 <wkingnumber2@gmail.com>

# Summary of Changes

Log out the user and shutdown the agent if the app has been in the background (or the phone has been asleep) for 5 minutes
image

# Related Issues

N/A

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: core/App/contexts/auth.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ export interface AuthContext {
   checkPIN: (pin: string) => Promise<boolean>
   convertToUseBiometrics: () => Promise<boolean>
   getWalletCredentials: () => Promise<WalletSecret | undefined>
+  wipeSavedWalletSecret: () => void
```
Comment (jleach) [inline core/App/contexts/auth.tsx]: better terminology for "wipe" like remove, clear or delete. I get wipe but maybe harder for non-english spears.

File: core/App/navigators/RootStack.tsx [hunk 2]
```diff
@@ -25,18 +27,65 @@ import { createDefaultStackOptions } from './defaultStackOptions'
...
+      setAppStateVisible(appState.current)
+    })
+  }, [])
+
+  useEffect(() => {
+    if (appStateVisible.match(/active/) && prevAppStateVisible.match(/inactive|background/) && backgroundTime) {
+      // prevents the user from being locked out during metro reloading
+      setPrevAppStateVisible(appStateVisible)
+      //lock user out after 5 minutes
+      const MS_PER_MINUTE = 60000
```
Comment (jleach) [inline core/App/navigators/RootStack.tsx]: Prefer cammelCaseForVars as its what we generally use throughout the app.

File: core/App/screens/Splash.tsx [hunk 3]
```diff
@@ -141,29 +142,33 @@ const Splash: React.FC<SplashProps> = (props: SplashProps) => {
...
-        )
-
-        const wsTransport = new WsOutboundTransport()
-        const httpTransport = new HttpOutboundTransport()
-
-        newAgent.registerOutboundTransport(wsTransport)
-        newAgent.registerOutboundTransport(httpTransport)
-
-        await newAgent.initialize()
+        let newAgent = agent
```
Comment (jleach) [inline core/App/screens/Splash.tsx]: Does this do anything? I think objects are passed by ref so I'm not sure it will mint a new object.

**Review by jleach — CHANGES_REQUESTED** [review]
A few comments for consideration.

File: core/App/navigators/RootStack.tsx [hunk 4]
```diff
@@ -23,13 +26,60 @@ import { createDefaultStackOptions } from './defaultStackOptions'
...
+      setAppStateVisible(appState.current)
+    })
+  }, [])
+
+  useEffect(() => {
+    if (appStateVisible.match(/active/) && prevAppStateVisible.match(/inactive|background/) && backgroundTime) {
+      // prevents the user from being locked out during metro reloading
+      setPrevAppStateVisible(appStateVisible)
+      //lock user out after 5 minutes
+      const msPerMinute = 60000
```
Comment (jleach) [inline core/App/navigators/RootStack.tsx]: Can you move this to `constants.ts` I think it's something that many deployments will want to customize.  Would be handy for `0` to mean dont-lock-ever-any-timeout.

**Review by jleach — CHANGES_REQUESTED** [review]

File: core/App/navigators/RootStack.tsx (hunk 4 above)
Comment (wadeking98) [reply core/App/navigators/RootStack.tsx]: done

**Review by jleach — APPROVED** [review]

**Pull request merged**